examples/vmod_be/src/lib.rs

```diff
@@ -1,42 +1,46 @@
-#![allow(non_camel_case_types)]
-
-varnish::boilerplate!();
-
 use std::error::Error;
 
-use varnish::vcl::backend::{Backend, Serve, Transfer, VCLBackendPtr};
+use varnish::vcl::backend::{Serve, Transfer};
 use varnish::vcl::ctx::Ctx;
 
-varnish::vtc!(test01);
+#[varnish::vmod]
+mod be {
+    use varnish::vcl::backend::{Backend, VCLBackendPtr};
+    use varnish::vcl::ctx::Ctx;
 
-// parrot is our VCL object, which just holds a rust Backend,
-// it only needs two functions:
-// - new(), so that the VCL can instantiate it
-// - backend(), so that we can produce a C pointer for varnish to use
-pub struct parrot {
-    be: Backend<Sentence, Body>,
-}
+    use super::{Body, Sentence};
 
-impl parrot {
-    pub fn new(ctx: &mut Ctx, vcl_name: &str, to_repeat: &str) -> Result<Self, Box<dyn Error>> {
-        // to create the backend, we need:
-        // - the vcl context, that we just pass along
-        // - the vcl_name (how the vcl writer named the object)
-        // - a struct that implements the Serve trait
-        let be = Backend::new(
-            ctx,
-            vcl_name,
-            Sentence {
-                v: Vec::from(to_repeat),
-            },
-            false,
-        )?;
-
-        Ok(parrot { be })
+    // parrot is our VCL object, which just holds a rust Backend,
+    // it only needs two functions:
+    // - new(), so that the VCL can instantiate it
+    // - backend(), so that we can produce a C pointer for varnish to use
+    #[allow(non_camel_case_types)]
+    pub struct parrot {
+        be: Backend<Sentence, Body>,
     }
 
-    pub fn backend(&self, _ctx: &Ctx) -> VCLBackendPtr {
-        self.be.vcl_ptr()
+    impl parrot {
+        pub fn new(ctx: &mut Ctx, to_repeat: &str) -> Result<Self, String> {
+            // to create the backend, we need:
+            // - the vcl context, that we just pass along
+            // - the vcl_name (how the vcl writer named the object)
+            // - a struct that implements the Serve trait
+            let be = Backend::new(
+                ctx,
+                "TODO", // FIXME: This should be the vcl_name param that currently is not forwarded
```

Comment:
that is important because that allows us to name the backend/processors we create

Reply:
Implemented

Reply:
not that it's not really magic or anything, in the C world, `ctx` and `vcl_name` are the first arguments of the constructor. Maybe it would be easier to just force them to be in the `rust` world too?